### ISSUE TITLE
Provide way to manually instrument WinUIUnhandledExceptionHandler

### ISSUE THREAD

**Issue by jamescrosswell** [getsentry/sentry-dotnet#3561]
https://github.com/getsentry/sentry-dotnet/blob/79b1dc396b86f555f7135f8b3c19907a19f00fac/src/Sentry/Integrations/WinUIUnhandledExceptionIntegration.cs#L104-L105

- [ ] Code changes
- [ ] Docs

## Code changes

From [here](https://github.com/getsentry/sentry-dotnet/issues/2778#issuecomment-1814073479), Sentry 3.x has the reflection equivalient of:
- https://github.com/getsentry/sentry-dotnet/blob/343d921c78e9cd1bd47734cc70be95a34cef8fe7/src/Sentry/Integrations/WinUIUnhandledExceptionIntegration.cs#L81

[The exception handler it wires up](https://github.com/getsentry/sentry-dotnet/blob/343d921c78e9cd1bd47734cc70be95a34cef8fe7/src/Sentry/Integrations/WinUIUnhandledExceptionIntegration.cs#L96C1-L135C6) is currently private though. To make this easier for users, we need to expose some way for them to be able to wire this up via our public API.

## Docs

Once the code changes have been made, we should update the docs to show how people can manually wire up the exception handler that we provide.

Cross reference docs with [this comment](https://github.com/getsentry/sentry-dotnet/issues/2778#issuecomment-1829559140).

**Comment by jamescrosswell** [comment]
Although it's possible to create an API to let SDK users manually attach Sentry's unhandled exception handler for WinUI 3 applications, there are a couple of red flags in the code that make me think we don't want to do this, The main one is this:
https://github.com/getsentry/sentry-dotnet/blob/cf26d464a1608b5f75165d88eb7449aea0e216ec/src/Sentry/Integrations/WinUIUnhandledExceptionIntegration.cs#L116-L117

Because we don't take a dependency on the WinUI3 packages we don't have access to some of the critical types like `Microsoft.UI.Xaml.UnhandledExceptionEventArgs`... which means that, even in our event handler, we're using reflection to read out the properties of the eventArgs. 

Without using reflection then, I think we keep coming back to the same conclusion (that we would need to add a Sentry.WinUI3 package or something to properly support WinUI 3 applications without people having to wire up the event handler by copying/pasting code from our documentation). 

Building a Sentry.WinUI3 package would be challenging however. For one thing, currently we build and publish all of our NuGet packages from our macOS runner on GitHub... I don't know how we could do that for a package with WinUI3 dependencies, so the process that we use for release builds would start to get very complicated and more fragile.

**Comment by bruno-garcia** [comment]
> Building a Sentry.WinUI3 package 

Wouldn't it be `Sentry.UI.Xaml`? As it would depend on `Microsoft.UI.Xaml` 

> challenging however. For one thing, currently we build and publish all of our NuGet packages from our macOS runner on GitHub.

It seems like it carries a bunch of native libs but for building/packaging purposes I imagine it would build on Mac if we left it at `netstandard2.X`, no? 

If we had to build it on windows (for packaging purposes, I understand we need for running tests), we would need to do what we did on Unity. Where CI builds on the platforms that are needed. Then assembles the package once all builds are ready and artifacts available.
It would definitely be time consuming to set that up though.
